puzzles/models.py

```diff
@@ -142,6 +158,64 @@ def clear_answer(self, guess):
         self.save()
 
 
+# Hooks for syncing metas and tags
+
```

Comment:
This file is getting rather large. I think we should separate tag specific logic into a separate file.

I think it's fine to do in a separate PR.